### PR TITLE
this is for psroi_pool op, test=develop

### PR DESCRIPTION
**Pull request by SunGaofeng**
This is for PSROIPooling.

### CODE REVIEW

File: paddle/fluid/operators/psroi_pool_op.h [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+    framework::Tensor rois_batch_id_list;
+    rois_batch_id_list.Resize({rois_num});
+    int* rois_batch_id_data =
+        rois_batch_id_list.mutable_data<int>(ctx.GetPlace());
+
+    auto rois_lod = rois->lod().back();
+    int rois_batch_size = rois_lod.size() - 1;
+    PADDLE_ENFORCE_EQ(
+        rois_batch_size, batch_size,
+        "the rois_batch_size and imgs batch_size should be the same.");
```
Comment (heavengate) [inline paddle/fluid/operators/psroi_pool_op.h]: imgs -> feature maps / inputs ?

File: python/paddle/fluid/layers/nn.py [hunk 2]
```diff
@@ -9083,3 +9084,49 @@ def get_tensor_from_selected_rows(x, name=None):
...
+        Variable: ${out_comment}.
+
+    Examples:
+        .. code-block:: python
+
+            pool_out = fluid.layers.psroi_pool(input=x, rois=rois, 490, 1.0, 7, 7)
+    """
+    helper = LayerHelper('psroi_pool', **local())
+    # check attrs
+    if isinstance(output_channels, int) is False:
```
Comment (heavengate) [inline python/paddle/fluid/layers/nn.py]: if not isinstance(..):

**Review by heavengate — COMMENTED** [review]
python Api test in test_layers.py should be added.

File: python/paddle/fluid/layers/nn.py (hunk 2 above)
Comment (SunGaofeng) [reply python/paddle/fluid/layers/nn.py]: Yes, modified

File: python/paddle/fluid/layers/nn.py [hunk 3]
```diff
@@ -9083,3 +9084,50 @@ def get_tensor_from_selected_rows(x, name=None):
         outputs={'Out': out},
         attrs={})
     return out
+
+
+def psroi_pool(input, rois, output_channels, spatial_scale, pooled_height,
+               pooled_width):
```
Comment (qingqing01) [inline python/paddle/fluid/layers/nn.py]: add name at last, better to make API consistent.

```
pooled_width, name=None):
```

**Review by qingqing01 — COMMENTED** [review]
LGTM

File: paddle/fluid/operators/psroi_pool_op.h (hunk 1 above)
Comment (SunGaofeng) [reply paddle/fluid/operators/psroi_pool_op.h]: Yes, modified to Input(X)

File: python/paddle/fluid/layers/nn.py (hunk 3 above)
Comment (SunGaofeng) [reply python/paddle/fluid/layers/nn.py]: done

**Review by qingqing01 — APPROVED** [review]
LGTM.

**Review by panyx0718 — APPROVED** [review]
LG API

**Pull request merged**